### ISSUE TITLE
Installation problems

### ISSUE THREAD

**Issue by lbusett** [ropensci/MODIStsp#182]
(Received by mail from user Johannes Piipponen and moved here)

Hi, I have confronted all the issues with MODIStsp package.
Anyway, since I have to work a lot with MODIS data in the future I guess this package is essential. Also my research team would use it if it appears to be useful.

 If you just have time to help me with  some issues:

1)  installation of GDAL fails - do I need to install python before installing GDAL?
2)  since install_github("ropensci/MODIStsp")  and install_github("ropensci/MODIStsp") won`t work properly, is the use of link http://ftp.gnome.org/pub/gnome/binaries/win64/gtk+/2.22/gtk+-bundle_2.22.1-20101229_win64.zip only way to proceed?

3)  if I get GDAL and libraries gWidgetsRGtk2 and ropensci/MODIStsp to work, is everything more or less functional after that?

4)  is it possible to build the package so that installation would be a little bit easier? This is not a complain, I just thought many users would stop trying at this point when the package cannot be loaded too easily..

**Comment by lbusett** [comment]
Hi Johannes, 

thanks for your interest in our package. I'll try to reply inline:

> 1. installation of GDAL fails - do I need to install python before installing GDAL?

Difficult to reply without further info. What OS are you using? Also, did you already try instructions reported here: https://ropensci.github.io/MODIStsp/articles/installation.html#installing-r-and-gdal

>2. since install_github("ropensci/MODIStsp") and install_github("ropensci/MODIStsp") won`t work properly, is the use of link http://ftp.gnome.org/pub/gnome/binaries/win64/gtk+/2.22/gtk+-bundle_2.22.1-20101229_win64.zip only way to proceed?

1. Is there a reason for not installing the CRAN version using `install.packages("MODIStsp")`
2. What error do you get from `install_github("ropensci/MODIStsp")`

> 3. if I get GDAL and libraries gWidgetsRGtk2 and ropensci/MODIStsp to work, is everything more or less functional after that?

Yes

> 4. is it possible to build the package so that installation would be a little bit easier? This is not a complain, I just thought many users would stop trying at this point when the package cannot be loaded too easily..

Not at the moment, sorry. I plan to substitute the gwidgets interface with a Shiny-based one in the future, but I do not know when I will be able to do it. 

Lorenzo

**Comment by jpiippon** [comment]
Hi and thanks for the reply!
1) I have Windows 64 bit operating system, and I followed these vignette instructions step by step: 
https://cran.r-project.org/web/packages/MODIStsp/vignettes/MODIStsp.pdf
Express Desktop Install and then I unchecked the installation of Qgis and GRASS GIS:


2) Well, I think also CRAN version will do just fine so maybe I try with it.
 I was able to install package ‘gWidgetsRGtk2’ but not "RGtk2" : it gives error
"Error in install.packages : ERROR: failed to lock directory" 

After I deleted 00Lock from packages installation of ‘gWidgetsRGtk2’  was possible. Now idea where the 00Lock came from but it releted to somehow to these abovementioned installations.

After that CRAN version "MODIStsp" worked! But still troubles with installing GDAL..

3) Great!
4) Thanks for you effort anyway - I think this package will be useful!
